src/index.js

```diff
@@ -470,10 +478,11 @@ export default class HyperScreen extends React.Component {
    * preload screen and URL to load.
    */
   componentWillReceiveProps(nextProps) {
-    const newUrl = nextProps.navigation.state.params.url;
-    const oldUrl = this.props.navigation.state.params.url;
-    const newPreloadScreen = nextProps.navigation.state.params.preloadScreen;
-    const oldPreloadScreen = this.props.navigation.state.params.preloadScreen;
+    const { params } = this.getNavigationState();
+    const newUrl = params.url;
+    const oldUrl = params.url;
+    const newPreloadScreen = params.preloadScreen;
+    const oldPreloadScreen = params.preloadScreen;
```

Comment:
The `oldPreloadScreen` and `newPreloadScreen` are pointing to the same `params.preloadScreen`.

Reply:
addressed in f5396f3